rust/ql/lib/codeql/rust/controlflow/internal/ControlFlowGraphImpl.qll

```diff
@@ -661,7 +704,9 @@ module PatternTrees {
 
   class LiteralPatTree extends LeafTree, LiteralPat { }
 
-  class MacroPatTree extends LeafTree, MacroPat { } // todo
+  class MacroPatTree extends PreOrderPatTree, MacroPat {
```

Comment:
We need a case for `MatroPat` in `isIrrefutablePattern` and `condPropagateExpr`.

Reply:
Does this look good for `isIrrefutablePattern`? https://github.com/github/codeql/pull/17746

Reply:
Shouldn't `condPropagateExpr` be covered already, `MacroPat` because is defined as a `PreOrderPatTree`. Or does it need something else?